tsdb/engine/tsm1/tombstone.go

```diff
@@ -155,28 +155,30 @@ func (t *Tombstoner) writeTombstone(tombstones []Tombstone) error {
 
 func (t *Tombstoner) readTombstone() ([]Tombstone, error) {
 	f, err := os.Open(t.tombstonePath())
+	if os.IsNotExist(err) {
+		return nil, nil
+	} else if err != nil {
+		return nil, err
+	}
 	defer f.Close()
 
-	if !os.IsNotExist(err) {
-		var b [4]byte
-		_, err := f.Read(b[:])
-		if err != nil {
-			// Might be a zero length file which should not exist, but
-			// an old bug allowed them to occur.  Treat it as an empty
-			// v1 tombstone file so we don't abort loading the TSM file.
-			return t.readTombstoneV1(f)
-		}
+	var b [4]byte
+	_, err = f.Read(b[:])
```

Comment:
I agree with @benbjohnson's point. At least theoretically, you may get a partial read.

E.g., if the fie is just 2 bytes `ab`, doing:

```Go
var b [4]byte
_, err = f.Read(b[:]
```

May result in 1 byte being read and nil error being returned. To get the 2nd byte, you'd have to call `Read` again.

> Read reads **up to** len(p) bytes into p. It returns the number of bytes read (0 <= n <= len(p)) and any error encountered.
>
> Implementations of Read are **discouraged** from returning a zero byte count with a nil error, except when len(p) == 0. Callers should treat a return of 0 and nil as indicating that nothing happened; in particular it does not indicate EOF.

Emphasis added is mine.